doc/mysqlnd_azure_log.md

```diff
@@ -0,0 +1,124 @@
+# Diagnostic Log Configuration
+
+Log down some runtime information from mysqlnd\_azure.
+
+## Configuration
+
+### Log destination supported (mysqlnd\_azure.logOutput)
+
+**Notice:**
+For all the file paths mentioned below, except Apache and php-fpm's system error log, it need to be configured with right user permission, e.g. www-data for Apache. For relative file path, when working in cli, the father path is the current working directoy, when working with Apache, the father path is /.
+
+We provide multiple options for log destination you may want to choose. 
+
+Name |mysqlnd\_azure.logOutput
+:----- | :------
+Comment | Destination for log
+Type | Enumeration
+Accepted Value | [ 0\| 1\| 2 \| 4 ]
+Default | 0 (No log)
+Dynamic | No
+
+
+Value | Behavior | Sample log
+:----- | :------ | :----- 
+0 | Like logLevel control, it will turn off the log. No log will be print out. | 
+1 | Print logs using PHP error log API. <br/> - PHP CLI: Print to console. <br/> - Work scenario with Apache/Nginx: Will be discussed in seperated section below. |  Warning: mysqli_real_connect(): [2020-07-20 14:09:03] [MYSQLND_AZURE] [ERROR] CLIENT_SSL is not set when mysqlnd_azure.enableRedirect is ON in mysqli_basic.php on line 6
+2 | Print logs to a pre-defined file by mysqlnd\_azure's config mysqlnd\_azure.logfilePath=xx.log. | [MYSQLND_AZURE] [ERROR] CLIENT_SSL is not set when mysqlnd_azure.enableRedirect is ON
+4 | Print logs using stderr. <br/> - PHP CLI: Print to console. <br/> - Work scenario with Apache/Nginx: Will be discussed in seperated section below. | [MYSQLND_AZURE] [ERROR] CLIENT_SSL is not set when mysqlnd_azure.enableRedirect is ON
+
+#### Detailed behavior for logOutput1 and logOutput4
+- **For mod-php and Apache (take php7.2 as example)**
+
+Config file | Config options | Usage
+:----- | :------ | :----- 
+/etc/php/7.2/apache2/php.ini | error_log <br/> log_error <br/> display_errors   | This file is Apache mod_php's config file s <br/> -log_error: logOutput1 is only available if this is on. <br/> error_log: where logOutput1 goes into. <br/> --1.	error_log=syslog:  logOutput1 is not available with this option. <br/> --2.	error_log=A_file_path. It will take priority of apache’s error log setting. <br/> display_errors: Whether print logOutput1 to a client web page.
```

Comment:
```suggestion
/etc/php/7.2/apache2/php.ini | error_log <br/> log_error <br/> display_errors   | This file is Apache mod_php's config file<br/> -log_error: logOutput=1 is only available if this is on. <br/> error_log: where logOutput=1 goes into. <br/> (1)	error_log=syslog:  logOutput= 1 is not available with this option. <br/>(2) error_log=A_file_path. It will take priority of Apache’s error log setting. <br/> display_errors: Whether print logOutput=1 to a client web page.
```